opencog/rule-engine/Rule.cc

```diff
@@ -484,14 +483,20 @@ std::string Rule::to_string(const std::string& indent) const
 	return ss.str();
 }
 
-Rule Rule::rand_alpha_converted() const
+Rule Rule::rand_alpha_converted(const Variables& vardecl) const
 {
 	// Clone the rule
 	Rule result = *this;
 
 	// Alpha convert the rule
 	result.set_rule(_rule->alpha_convert());
 
+	// Check for the small chance of still having name collisions
+	while(is_disjoint(_rule->get_variables().varset, vardecl.varset))
+	{
+		result.set_rule(_rule->alpha_convert());
+	}
```

Comment:
You can save the previous conversion by using a do...while loop.

Also, the condition is wrong, it should be the varset of result, not _rule.